content/blog/2022-07-26-schema-evolution-glue-msk/index.md

```diff
@@ -0,0 +1,861 @@
+   [AvroDeserializer][DEBUG] Finished de-serializing Avro message
+   [GlueSchemaRegistryDeserializerFactory][DEBUG] Returning Avro de-serializer instance from GlueSchemaRegistryDeserializerFactory
+   [GlueSchemaRegistryDefaultCompression][DEBUG] Decompression :: Compressed record length: 0KB
+   [GlueSchemaRegistryDefaultCompression][DEBUG] Decompression :: Decompressed record length: 0KB
+   [AvroDeserializer][DEBUG] Length of actual message: 121
+   [AvroDeserializer][DEBUG] Finished de-serializing Avro message
+   ```
+
+## Schema Evolution
+In the course of this tutorial, we have implemented a Kafka producer and a consumer which integrate with the Glue Schema Registry.
+But the full potential of the Glue Schema Registry is unlocked when performing a schema evolution, i.e. whzen running producers and consumers with a new version of an already registered schema.
+Therefore, we will run a few more interesting scenarios to illustrate the benefits of the Schema Registry:
+1. Running a producer which automatically registers a new, compatible schema version.
+2. Running a producer which is rejected when trying to register a new, incompatible schema version.
+3. Running an old consumer which is rejected since it is not compatible with the newly registered schema version.
+4. Running an updated consumer which can consume the new schema / records.
+
+### 1. Producer registering a new schema version
```

Comment:
I'm not sure if its ideal to use numbers in headings.

Reply:
I don't have a strong opinion on it, I just wanted to directly associate the scenarios in the subsections with the ones shortly outlined above. What do you think? Should I remove the numbers?